### PR TITLE
chore(snapshots): openapi translations for transactions and state

### PR DESCRIPTION
**Pull request by dsharma-dc**

### CODE REVIEW

**Comment by dsharma-dc** [comment]
```
{
  "definition": {
    "metadata": {
      "txn_id": "1",
      "transactions": {
        "1": [
          {
            "uuid": "a25edd55-2816-43cf-8b6a-655acbcc308c",
            "source_id": "88ec6da9-faed-4c63-be10-4f5915ae6a3c",
            "status": "Created"
          }
        ]
      }
    },
    "spec": {
      "uuid": "15823425-41fa-434a-9efd-a356b70b5d7c",
      "source_volume": "ec4e66fd-3b33-4439-b504-d49aba53da26"
    }
  },
  "state": {
    "uuid": "15823425-41fa-434a-9efd-a356b70b5d7c",
    "size": 12582912,
    "source_volume": "ec4e66fd-3b33-4439-b504-d49aba53da26",
    "creation_timestamp": "",
    "clone_ready": false,
    "replica_snapshots": [
      {
        "uuid": "a25edd55-2816-43cf-8b6a-655acbcc308c",
        "source_id": "ec4e66fd-3b33-4439-b504-d49aba53da26",
        "creation_timestamp": "1970-01-01T00:00:00Z",
        "size": 12582912,
        "referenced_size": 12582912,
        "state": "Online"
      }
    ]
  }
}
```

**Comment by dsharma-dc** [comment]
bors try

File: control-plane/rest/service/src/v0/snapshots.rs [hunk 1]
```diff
@@ -165,7 +178,47 @@ pub fn to_models_volume_snapshot(snap: &VolumeSnapshot) -> models::VolumeSnapsho
...
+            snap.state()
+                .repl_snapshots()
+                .iter()
+                .map(to_models_replica_snapshot_state)
+                .collect::<Vec<_>>(),
         ),
     }
 }
+
+pub fn to_models_replica_snapshot(repl_snap: &ReplicaSnapshot) -> models::ReplicaSnapshot {
```
Comment (tiagolobocastro) [inline control-plane/rest/service/src/v0/snapshots.rs]: these don't need to be pub

**Review by tiagolobocastro — APPROVED** [review]

**Comment by dsharma-dc** [comment]
bors merge

**Comment by dsharma-dc** [comment]
bors merge

**Pull request merged**